amlb/datasets/openml.py

```diff
@@ -139,133 +127,154 @@ def get_values(f):
+
+
+class ArffSplitter(DataSplitter[str]):
+    format = 'arff'
+
+    def __init__(self, ds: OpenmlDataset):
+        super().__init__(ds)
+
+    def split(self) -> Tuple[str, str]:
+        train_path, test_path = self.ds._get_split_paths(".arff")
+        if not os.path.isfile(train_path) or not os.path.isfile(test_path):
+            X, *_ = self.ds._oml_dataset.get_data(dataset_format='dataframe')
+            train, test = X.iloc[self.train_ind, :], X.iloc[self.test_ind, :]
+            name_template = "{name}_{{split}}_{fold}".format(name=self.ds._oml_dataset.name, fold=self.ds.fold)
+            self._save_split(train, train_path, name_template.format(split="train"))
+            self._save_split(test, test_path, name_template.format(split="test"))
+        return train_path, test_path
+
+    @profile(logger=log)
+    def _save_split(self, df, path, name):
+        log.debug("Saving %s split dataset to %s.", name, path)
+        with open(path, 'w') as file:
+            description = f"Split dataset file generated by automlbenchmark from OpenML dataset openml.org/d/{self.ds._oml_dataset.dataset_id}"
+            attributes = [(c,
+                           ('INTEGER' if pat.is_integer_dtype(dt)
+                            else 'REAL' if pat.is_float_dtype(dt)
+                            else 'NUMERIC' if pat.is_numeric_dtype(dt)
+                            else self._get_categorical_values(c) if pat.is_categorical_dtype(dt)
+                            else 'STRING'
+                           ))
+                          for c, dt in zip(df.columns, df.dtypes)]
+            arff.dump({
+                'description': description,
+                'relation': name,
+                'attributes': attributes,
+                'data': df.values
+            }, file)
+
+    def _get_categorical_values(self, col):
+        feat = next((f for f in self.ds._oml_dataset.features.values() if f.name == col), None)
+        return feat.nominal_values if feat is not None else None
+
+
+class CsvSplitter(DataSplitter[str]):
+    format = 'csv'
+
+    def __init__(self, ds: OpenmlDataset):
+        super().__init__(ds)
+
+    def split(self) -> Tuple[str, str]:
+        train_path, test_path = self.ds._get_split_paths(".csv")
+        if not os.path.isfile(train_path) or not os.path.isfile(test_path):
+            X, *_ = self.ds._oml_dataset.get_data(dataset_format='dataframe')
+            train, test = X.iloc[self.train_ind, :], X.iloc[self.test_ind, :]
+            train.to_csv(train_path, header=True, index=False)
+            test.to_csv(test_path, header=True, index=False)
+        return train_path, test_path
+
+
+__data_splitters__ = [ArraySplitter, DataFrameSplitter, ArffSplitter, CsvSplitter]
+__supported_file_formats__ = ['arff', 'csv']
+
+
+def _get_data_splitter_cls(split_format='array'):
+    return next(ds for ds in __data_splitters__ if ds.format == split_format)
```

Comment:
It's not wrong, but a `StopIteration` error might be a little hard understand (especially for someone integrating with the framework requesting an unavailable split format). Maybe add a friendlier message?

```python
supported = {ds.format: ds for ds in __data_splitters__}
if split_format not in supported :
     raise ValueError(f"No splitter for format '{split_format}' available, only {list(supported)} are supported.")
return supported[split_format]
```